addons/mail/controllers/mail.py

```diff
@@ -165,3 +165,26 @@ def mail_action_view(self, model=None, res_id=None, access_token=None, **kwargs)
             except ValueError:
                 res_id = False
         return self._redirect_to_record(model, res_id, access_token, **kwargs)
+
+    @http.route('/mail/unfollow', type='http', auth='public')
+    def mail_action_unfollow(self, model, res_id, pid, token, **kwargs):
+        comparison, record, redirect = MailController._check_token_and_record_or_redirect(model, int(res_id), token)
+        if not comparison or not record:
+            raise AccessError(_('Non existing record or wrong token.'))
+
+        pid = int(pid)
+        record_sudo = record.sudo()
+        if pid in record_sudo.message_partner_ids.ids:
+            partner = request.env['res.partner'].sudo().browse([pid])
```

Comment:
we can directly give the integer (check also other place in the PR)
```python
partner = request.env['res.partner'].sudo().browse([pid])
```

Reply:
thanks for the tip.